src/main/java/com/internet/shop/util/HashUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.internet.shop.util;
+
+import com.internet.shop.model.User;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final String ALGORITHM = "SHA-512";
+
+    public static byte[] getSalt() {
+        SecureRandom random = new SecureRandom();
+        byte[] salt = new byte[16];
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digests = messageDigest.digest(password.getBytes());
+            for (byte b : digests) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Can't find algorithm " + ALGORITHM + e);
+        }
+        return hashedPassword.toString();
+    }
+
+    public static boolean isPassValid(String password, User user) {
+        return hashPassword(password, user.getSalt())
+                .equals(user.getPassword());
+    }
```

Comment:
let's make this method private in AuthenticationServiceImpl

Reply:
Done